src/notebookConcatDocument.ts

```diff
@@ -711,8 +717,8 @@ export class NotebookConcatDocument implements vscode.TextDocument, vscode.Dispo
         const lines = splitLines(text);
         let spans: NotebookSpan[] = [];
 
-        // If this the absolute first cell, add the header spans in
-        if (offset == 0) {
+        // If this the absolute first cell, add the header spans in (skip if input box)
+        if (offset == 0 && !cellUri.scheme.includes(InteractiveInputScheme)) {
```

Comment:
Order may have changed that VS code sends the 'open' for the interactive window parts, but the 'startup' commands have to be attached to the first 'real' cell run or they're not contiguous. The contiguous part is necessary for the change algorithm to work. So this was breaking stuff too.